build/org.eclipse.cdt.managedbuilder.core.tests/tests/org/eclipse/cdt/managedbuilder/core/tests/CompilationDatabaseGenerationTest.java

```diff
@@ -167,4 +170,43 @@ public void restoreDefaultForGenerateFile() {
 				"org.eclipse.cdt.managedbuilder.ui");
 		preferenceStore.setToDefault(CommonBuilder.COMPILATION_DATABASE_ENABLEMENT);
 	}
+
+	@Test
+	public void testCompilerPath() throws Exception {
+		setWorkspace("regressions");
+		final IProject app = loadProject("helloworldC");
+		setGenerateFileOptionEnabled(true);
+		app.build(IncrementalProjectBuilder.FULL_BUILD, null);
+		IFile commandsFile = app.getFile("Debug/compile_commands.json");
+
+		if (commandsFile.exists()) {
+			try (FileReader reader = new FileReader(commandsFile.getLocation().toFile())) {
+				Gson gson = new Gson();
+				JsonArray jsonArray = gson.fromJson(reader, JsonArray.class);
+				for (JsonElement element : jsonArray) {
+					CompilationDatabaseInformation compileCommand = gson.fromJson(element,
+							CompilationDatabaseInformation.class);
+
+					String command = compileCommand.command();
+					String compilerPath;
+					if (command.startsWith("\"")) {
+						// Linux format with quotes around paths
+						int endIndex = command.indexOf("\"", 1); // find closing quote
+						compilerPath = command.substring(1, endIndex);
+					} else {
+						// Windows format with direct path
```

Comment:
I don't understand this else branch. In the code change you seem to always surround with `"`.

For this test can you use CommandLineUtil.argumentsToArray and then verify `result[0]` is an absolute path?